src/components/NewSpot/component.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { Circle, Map, TileLayer } from "react-leaflet";
+
+import { useDependency } from "../../di";
+
+export interface NewSpotProps {}
+
+export const NewSpot: React.FC<NewSpotProps> = () => {
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+
+  const searchParams = new URLSearchParams(window.location.search);
+  const latitudeStr = searchParams.get("latitude");
+  const longitudeStr = searchParams.get("longitude");
+
+  const useCase = useDependency("createSpot");
+
+  if (!latitudeStr || !longitudeStr) {
+    throw new Error("Missing required location.");
+  }
+
+  const latitude = Number(latitudeStr);
+  const longitude = Number(longitudeStr);
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    useCase.execute(
+      title,
+      {
+        latitude,
+        longitude
+      },
+      description
+    );
+  };
+
+  return (
+    <div>
+      <Map
```

Comment:
As we saw on Saturday Leaflet seems pretty buggy on location so we might change the Map implementation. Thus it would be nice to abtract leaflet from these components

Reply:
@jordantaillefer @francoischalifour I guess we can merge this PR like this and work together on extracting a generic `Map` component to decouple our project from any map library